utils/sysfs/sysfs.go

```diff
@@ -335,25 +348,198 @@ func (fs *realSysFs) GetSystemUUID() (string, error) {
 	}
 }
 
-func (fs *realSysFs) IsCPUOnline(dir string) bool {
-	cpuPath := fmt.Sprintf("%s/online", dir)
-	content, err := ioutil.ReadFile(cpuPath)
+func (fs *realSysFs) IsCPUOnline(cpuPath string) bool {
+	return isCPUOnline(cpuPath)
+}
+
+// isCPUOnline verifies if a CPU in online. Takes path to /sys/devices/system/cpu/cpu*/ directory as a parameter.
+func isCPUOnline(cpuPath string) bool {
+	if isX86 && isZeroCPU(cpuPath) {
+		// It *is* possible to enable CPU hotplug for CPU0 but it requires setting a boot flag.
+		// It is assumed it's a rare situation.
+		return true
+	}
+
+	const isOnline = byte('1')
+	onlinePath := filepath.Join(cpuPath, "online")
+	onlineVal, err := ioutil.ReadFile(onlinePath)
+	// First, make sure that no error occurred.
 	if err != nil {
-		pathErr, ok := err.(*os.PathError)
-		if ok {
-			if errors.Is(pathErr.Unwrap(), os.ErrNotExist) && isZeroCPU(dir) {
-				return true
-			}
+		if !os.IsNotExist(err) {
+			// If online file exists but reading failed we assume that CPU is offline.
+			klog.V(2).Infof("Cannot determine CPU %s online state, assuming offline", cpuPath)
+			return false
 		}
-		klog.Warningf("unable to read %s: %s", cpuPath, err.Error())
+		// If online file does not exists then we should check one more location.
+		klog.V(2).Infof("%s not found, checking %s directory", onlinePath, filepath.Join(cpuPath, "hotplug"))
```

Comment:
regarding the log: `isOnlineHotplug` below actually checks  `filepath.Join(cpuPath, "hotplug", "state")` not `filepath.Join(cpuPath, "hotplug")` right? is the logged path incorrect?